src/irmin-pack/unix/pack_store.ml

```diff
@@ -156,6 +156,15 @@ struct
         Int63.pp off bytes_read;
     let hash = decode_bin_hash (Bytes.unsafe_to_string buf) (ref 0) in
     let kind = Pack_value.Kind.of_magic_exn (Bytes.get buf Hash.hash_size) in
+    (* This function is called on the parents of a commit when deserialising it.
+       Dangling_parent_commit are usually treated as Gced, expect here wherecin
+       order to correctly deserialise the commit, they are treated as preserved
+       (not gced) commits. *)
+    let kind =
+      if kind = Pack_value.Kind.Dangling_parent_commit then
+        Pack_value.Kind.Commit_v2
+      else kind
+    in
```

Comment:
I don't really like having this bit here because I think that `read_and_decode_entry_prefix` should strictly do what the name of the function imply.

What about instead moving this check to `key_of_offset`, just after `io_read_and_decode_entry_prefix` has been called?

Reply:
I moved it, thanks for the suggestion